### ISSUE TITLE
Validation feature request

### ISSUE THREAD

**Issue by bkamins** [JuliaData/CSV.jl#282]
I would like to put under consideration to add `validate` keyword to `CSV.File` that would turn on/off detailed validation of the file via `CSV.Validate`.

Additionally personally I would like to `validate` default to `true` as people should opt-out of validation if they need performance.

**Comment by bkamins** [comment]
Just to put an example of a problem I want to solve is that currently `CSV.read` has the following behavior:
```
julia> CSV.read(IOBuffer("a,b,c\n1\n2\n3"))
3×3 DataFrames.DataFrame
│ Row │ a       │ b       │ c       │
│     │ Int64⍰  │ Int64⍰  │ Int64⍰  │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1       │ 2       │ 3       │
│ 2   │ missing │ missing │ missing │
│ 3   │ missing │ missing │ missing │
```
which is caught only by `CSV.validate`.

**Comment by bkamins** [comment]
CC @nalimilan

**Comment by quinnj** [comment]
I can take a look at that specific issue, but I'd have preferred opening a new issue, since it's really a separate issue than the discussion around `validate`.

**Comment by quinnj** [comment]
@bkamins can you check your version please?
```julia
julia> using CSV, Test, Tables
[ Info: Recompiling stale cache file /Users/jacobquinn/.julia/compiled/v1.0/CSV/HHBkp.ji for CSV [336ed68f-0bac-5ca0-87d4-7b16caf5d00b]

julia> CSV.read(IOBuffer("a,b,c\n1\n2\n3"))
3×3 DataFrames.DataFrame
│ Row │ a      │ b       │ c       │
│     │ Int64⍰ │ Missing │ Missing │
├─────┼────────┼─────────┼─────────┤
│ 1   │ 1      │ missing │ missing │
│ 2   │ 2      │ missing │ missing │
│ 3   │ 3      │ missing │ missing │

```

**Comment by bkamins** [comment]
Confirmed - on `HHBkp` it works like you reported above. Seems that CSV.jl is a fast moving target 😄 - fortunately in a good direction.

**Comment by nalimilan** [comment]
Cool. So what's the cost/benefit tradeoff of `validate=true`?

**Comment by quinnj** [comment]
I believe with #395, we should now cover everything discussed here (i.e. `CSV.validate` is deprecated and `CSV.File`/`CSV.read` just do validation as they parse. You can choose to throw errors on malformed rows via `strict=true`, by default `strict=false`. There is also `silencewarnings=false`, so malformed rows will be skipped by default, but a warning will be printed)

**Comment by JockLawrie** [comment]
Great! The new interface is a joy, and the imminent overhaul looks fantastic. Eagerly awaiting the next release. Thanks for your hard work.

**Comment by quinnj** [comment]
Fixed in #395
